lib/grell/page_collection.rb

```diff
@@ -39,8 +42,13 @@ def add(page)
       # Although finding unique pages based on URL will add pages with different query parameters,
       # in some cases we do link to different pages depending on the query parameters like when using proxies
       new_url = @collection.none? do |collection_page|
-        collection_page.url.downcase == page.url.downcase
+        if @add_match_block
```

Comment:
Instead of this, you can have crawler to create a match_block with 
```
collection_page.url.downcase == page.url.downcase
```
then this class does never know about how to match and expects the match to come from whoever calls it.
Crawler will have the match_block passed to it or its default which is the full url